### PR TITLE
Cleaned up a bunch of tests, race conditions and logging

### PR DESCRIPTION
**Pull request by uglybugger**

### CODE REVIEW

File: src/Nimbus/BatchSendingExtensions.cs [hunk 1]
```diff
@@ -9,12 +8,24 @@ public static class BatchSendingExtensions
     {
         public static Task SendAll(this IBus bus, IEnumerable<IBusCommand> busCommands)
         {
-            return Task.Run(async () => { await Task.WhenAll(busCommands.Select(e => bus.Send(e))); });
+            return Task.Run(async () =>
+                                  {
+                                      foreach (var c in busCommands)
+                                      {
+                                          await bus.Send(c);
```
Comment (robdmoore) [inline src/Nimbus/BatchSendingExtensions.cs]: Can that line throw an exception?

Comment (KodrAus) [inline src/Nimbus/BatchSendingExtensions.cs]: Also, is there a reason not to use `ConfigureAwait(false)` here and in the `PublishAll` method?

Comment (uglybugger) [inline src/Nimbus/BatchSendingExtensions.cs]: Re "can it throw?": Yep. The batch sending extensions are a bit of a stop-gap at the moment.

Comment (uglybugger) [inline src/Nimbus/BatchSendingExtensions.cs]: Re ConfigureAwait: Good point. I've just added that now in 333c840.

**Pull request merged**